algorithms/CPlusPlus/Backtracking/n-queens.cpp

```diff
@@ -0,0 +1,71 @@
+/* Code contributed by Devang Shah to MakeContributions/DSA GitHub repository */
+'''
+Below CPlusPlus code displays all possible solutions for a N-Queens problem. 
+Problem: Given a N x N chess board, arrange N queens in a way such that no two queens attack each other. Two queens are attacking each other if they are in same row, same column or diagonal
+Input: N
+Output: All solutions ( in grid format {Q - Queen, * - empty cell} ) of the N-queens problem
+Time Complexity: O(N!)
+'''
+
+#include <iostream>
+using namespace std;
+
+int a[30], count = 0;
```

Comment:
what if `n` is larger than 30 ?

Reply:
The time and space complexity of the problem is too large. I think OS will be unable to withstand n>15 also. After which it will kill the program execution. So n>30 is actually not feasible here. Anyways most of the competitive programming platforms have a constraint on N (between 1-10 only). 
Codechef: https://www.codechef.com/problems/TIC04
Leetcode: https://leetcode.com/problems/n-queens/
HackerEarth: https://www.hackerearth.com/practice/basic-programming/recursion/recursion-and-backtracking/practice-problems/algorithm/n-queensrecursion-tutorial/
![image](https://user-images.githubusercontent.com/80088008/186089707-79314932-2ca7-4c8c-bce1-a315cb0af486.png)